rollingpin/commands.py

```diff
@@ -0,0 +1,82 @@
+from abc import ABCMeta, abstractproperty
+
+
+class Command(object):
+    __metaclass__ = ABCMeta
+
+    CONTINUE = 1
+    SKIP_REMAINING = 2
+
+    def __init__(self, args=None):
+        if args is None:
+            self._args = []
+        else:
+            self._args = args
+
+    @abstractproperty
+    def name(self):
+        raise NotImplementedError
+
+    @property
+    def args(self):
+        return self._args
+
+    def add_argument(self, arg):
+        self._args.append(arg)
+
+    def cmdline(self):
+        return [self.name()] + self._args
+
+    def check_result(self, result):
+        return Command.CONTINUE
+
+    def __str__(self):
+        return "Command(name={}, args={})".format(self.name(), self._args)
+
+    def __repr__(self):
+        return self.__str__()
+
+
+class SynchronizeCommand(Command):
+    def name(self):
```

Comment:
funky, i've never used the `@abstractproperty` thing before. do the subclass implementations really not need the `@property` decorator themselves? that seems really confusing to read since you have to look at the base class to remember this doesn't want function call syntax.

fwiw, i'd usually do this sort of thing with just a class variable:

```python
class SynchronizeCommand(Command):
    name = "synchronize"
```